### PR TITLE
Feat/lifi widget

### PR DESCRIPTION
**Pull request by ry-animal**
## General Changes

- Adds LiFi Widget

https://github.com/seamless-protocol/interface/assets/118958436/87521f6f-250d-4f92-88e9-9fb0a00e75f1

### CODE REVIEW

File: src/components/LIFI/Widget.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use client';
+import { LiFiWidget, WidgetConfig, WidgetDrawer } from '@lifi/widget';
+import { useMemo, useRef } from 'react';
+
+import { WidgetEvents } from './WidgetEvents';
+
+export const Widget = () => {
+  const widgetConfig: WidgetConfig = useMemo(
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/components/LIFI/Widget.tsx:8]: copy pasta from their next example - removed

File: src/ui-config/menu-items/index.tsx [hunk 2]
```diff
@@ -29,6 +31,15 @@ export const navigation: Navigation[] = [
     title: t`Staking Farms`,
     dataCy: 'menuFarms',
   },
+  {
+    link: '',
+    title: t`Swap on to Base`,
+    dataCy: 'swapBase',
+    disabled: false,
+    action: () => {
+      window.open('https://app.seamless.finance/swap', '_blank');
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/ui-config/menu-items/index.tsx:40]: copilot - deleted

File: src/store/lifiSlice.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export const createLifiSlice: StateCreator<
+  RootStore,
+  [['zustand/subscribeWithSelector', never], ['zustand/devtools', never]],
+  [],
+  LifiSlice
+> = (set) => {
+  return {
+    isLifiWidgetOpen: false,
+    setLifiWidget(eventName: boolean) {
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/store/lifiSlice.ts:18]: following pattern from other slices

File: src/store/lifiSlice.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { StateCreator } from 'zustand';
+
+import { RootStore } from './root';
+
+export type LifiSlice = {
+  isLifiWidgetOpen: boolean;
+  setLifiWidget: (eventName: boolean) => void;
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/store/lifiSlice.ts:7]: setting the widget, follow on to externally control state when time allows

File: src/modules/dashboard/DashboardListTopPanel.tsx [hunk 5]
```diff
@@ -48,7 +49,10 @@ export const DashboardListTopPanel = ({
       />
 
       {(STAGING_ENV || ENABLE_TESTNET) && <FaucetButton />}
-      {!ENABLE_TESTNET && <BridgeButton bridge={bridge} />}
+      <Box sx={{ display: 'flex', gap: 4 }}>
+        {!ENABLE_TESTNET && <LifiBridgeButton bridge={bridge} />}
+        {!ENABLE_TESTNET && <BridgeButton bridge={bridge} />}
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardListTopPanel.tsx:54]: bridge = config for base

File: .env
[the remark this answers is not in the document]
Comment (ry-animal) [reply .env:1]: how does aave not have .env in gitignore, le sigh

File: src/modules/dashboard/DashboardListTopPanel.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardListTopPanel.tsx:54]: startIcon={}

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardListTopPanel.tsx:54]: according to design they are both base logos

File: next.config.js [hunk 6]
```diff
@@ -3,35 +3,41 @@ const withBundleAnalyzer = require('@next/bundle-analyzer')({
   enabled: process.env.ANALYZE === 'true',
 });
 
+// eslint-disable-next-line
+const withTM = require('next-transpile-modules')(['@lifi/widget']);
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply next.config.js:7]: "next": "12.1.1",

File: src/hooks/useAssetCaps.tsx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import { valueToBigNumber } from '@aave/math-utils';
-import { SxProps, Theme } from '@mui/system';
+import { SxProps, Theme } from '@mui/material';
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]: mui/system was what was causing compile issues

File: src/layouts/MainLayout.tsx [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
 import AnalyticsConsent from 'src/components/Analytics/AnalyticsConsent';
+import DynamicLifi from 'src/components/LIFI/DynamicLifi';
+import { useRootStore } from 'src/store/root';
 import { FORK_ENABLED } from 'src/utils/marketsAndNetworksConfig';
 
 import { AppFooter } from './AppFooter';
 import { AppHeader } from './AppHeader';
 
 export function MainLayout({ children }: { children: ReactNode }) {
+  const [lifiWidgetOpen] = useRootStore((state) => [state.isLifiWidgetOpen]);
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/layouts/MainLayout.tsx:12]: fix

File: src/components/LIFI/DynamicLifi.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import dynamic from 'next/dynamic';
+
+import { Widget } from './Widget';
+
+const DynamicLifi = dynamic(() => import('./Widget').then((module) => module.Widget), {
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/components/LIFI/DynamicLifi.tsx:5]: better perf, since we don't render until click and may never get clicked

File: src/components/LifiBridgeButton.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import { ExternalLinkIcon } from '@heroicons/react/outline';
+import { Button, SvgIcon, Typography } from '@mui/material';
+import { useRootStore } from 'src/store/root';
+
+import { NetworkConfig } from '../ui-config/networksConfig';
+
+export const LifiBridgeButton = ({ bridge }: Pick<NetworkConfig, 'bridge'>) => {
+  const [isLifiWidgetOpen, setLifiWidgetOpen] = useRootStore((state) => [
+    state.isLifiWidgetOpen,
+    state.setLifiWidget,
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/components/LifiBridgeButton.tsx:10]: fix

File: src/store/lifiSlice.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/store/lifiSlice.ts:18]: true, changed

File: src/layouts/components/NavItems.tsx [hunk 11]
```diff
@@ -63,7 +68,12 @@ export const NavItems = ({ setOpen }: NavItemsProps) => {
             ) : (
               <Button
                 component={Link}
-                onClick={() => handleClick(item.title, false)}
+                onClick={() => {
+                  item.dataCy === 'swapBase'
+                    ? setLifiWidgetOpen(true)
+                    : handleClick(item.title, false);
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/layouts/components/NavItems.tsx:74]: I started to but since it's just an array that's being imported here I would need to create something new with state when there is already state being used in the NavItems via 
```
const [isLifiWidgetOpen, setLifiWidgetOpen] = useRootStore((state) => [
    state.isLifiWidgetOpen,
    state.setLifiWidgetOpen,
  ]);
```

File: next.config.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (ry-animal) [reply next.config.js:7]: global css

File: src/hooks/useAssetCaps.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]: can try it again, the three components in the file were erroring due to incompatible sx types even though I didn't touch any of these files, maybe my tsc was just spazzing :
```
Type '{ borrowCap?: AssetCapData | undefined; debtCeiling?: AssetCapData | undefined; sx?: SxProps<Theme> | undefined; supplyCap: AssetCapData; icon: boolean | undefined; }' is not assignable to type 'AlertProps'.
  Types of property 'sx' are incompatible.
    Type 'import("/Users/ryan/repos/seashell/interface/node_modules/@mui/system/styleFunctionSx/styleFunctionSx").SxProps<import("/Users/ryan/repos/seashell/interface/node_modules/@mui/system/createTheme/createTheme").Theme> | undefined' is not assignable to type 'import("/Users/ryan/repos/seashell/interface/node_modules/@mui/material/node_modules/@mui/system/styleFunctionSx/styleFunctionSx").SxProps<import("/Users/ryan/repos/seashell/interface/node_modules/@mui/material/styles/createTheme").Theme> | undefined'.
      Type 'SystemCssProperties<Theme>' is not assignable to type 'SxProps<Theme> | undefined'.
        Type 'SystemCssProperties<Theme>' is not assignable to type 'CSSSelectorObjectOrCssVariables<Theme>'.
          Property 'inset' is incompatible with index signature.
            Type 'string | number | (string & {}) | SystemCssProperties<Theme> | CSSPseudoSelectorProps<Theme> | CSSSelectorObjectOrCssVariables<Theme> | ... 4 more ... | null' is not assignable to type 'SystemStyleObject<Theme> | CssVariableType | ((theme: Theme) => string | number | SystemStyleObject<Theme>)'.
              Type 'SystemCssProperties<Theme>' is not assignable to type 'SystemStyleObject<Theme> | CssVariableType | ((theme: Theme) => string | number | SystemStyleObject<Theme>)'.ts(2322)```

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]: Screenshot 2023-11-07 at 5 34 34 PM

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]: this also doesn't seem good, idk enough about material UI to know the diff but I assume @mui/material is the correct one

```
import { Button, Divider, Paper, Typography } from '@mui/material';
import { Box } from '@mui/system';
```

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]:
```
The @mui/material and @mui/system libraries are both part of the Material-UI (MUI) ecosystem, which is a popular React component library for building user interfaces with a Material Design look and feel. However, they serve different purposes within the Material-UI framework:

@mui/material:

@mui/material is the core library of Material-UI and provides the core set of Material Design components (e.g., buttons, cards, dialogs, etc.) along with styling support.
It includes pre-styled Material Design components and themes that you can use in your React applications.
@mui/material is typically used for building the user interface of your application, and it provides a set of ready-to-use components that follow the Material Design guidelines.
@mui/system:

@mui/system is a separate library within Material-UI that provides a styling system based on the System UI theme specification.
It is used for creating and applying custom styling to your Material-UI components and other components in your application.
@mui/system is a low-level styling library that allows you to define styles using a utility function called styled and provides a way to create your own themes and style components according to your design needs.
It is a more flexible and powerful styling solution for customizing the appearance of your Material-UI components and building unique designs.
In summary, @mui/material is primarily focused on providing pre-designed Material Design components and themes for your application's user interface. In contrast, @mui/system is a lower-level styling library that allows you to customize the styling of those components or style any component in your application in a more flexible and customized manner. You can use both libraries together to create a cohesive and visually appealing user interface with the benefits of Material Design while also having the flexibility to apply your own custom styles.```

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/hooks/useAssetCaps.tsx:2]: I'm guessing styling is being overridden, or maybe conflicting views on aave team, but I would assume things with /system are being overwritten but then again they are overriding styling and this is a /material below 
🤷

```
    <ToggleButtonGroup
      disabled={disabled}
      value={selectedTimeRange}
      exclusive
      onChange={handleChange}
      aria-label="Date range"
      sx={{
        height: '24px',
        '&.MuiToggleButtonGroup-grouped': {
          borderRadius: 'unset',
        },
        ...props.sx?.buttonGroup,
      }}
      ```

File: src/modules/dashboard/DashboardTopPanel.tsx [hunk 12]
```diff
@@ -237,6 +245,23 @@ export const DashboardTopPanel = () => {
               >
                 <Trans>Claim</Trans>
               </Button>
+              {connected && (
+                <AddTokenDropdown
+                  poolReserve={{
+                    ...reserves[1],
+                    symbol: 'OG Points',
+                    underlyingAsset: '0x5607718c64334eb5174CB2226af891a6ED82c7C6',
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardTopPanel.tsx:253]: Seems valid for a new story in the backlog, don't think it's realistic to grok how rewards get implemented in a lifi widget integration PR because the hardcoded version is an hour ask turnaround for user confusion using hardcoding

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardTopPanel.tsx:253]: RIP - guess it's as easy as using `selectedReward` 💀

File: src/components/transactions/ClaimRewards/ClaimRewardsModalContent.tsx [hunk 13]
```diff
@@ -125,6 +126,13 @@ export const ClaimRewardsModalContent = () => {
     }
   };
 
+  const addToken: ERC20TokenType = {
+    address: '0x5607718c64334eb5174CB2226af891a6ED82c7C6',
```
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/components/transactions/ClaimRewards/ClaimRewardsModalContent.tsx:130]: RIP - guess it's as easy as using `selectedReward` 💀

[the remark this answers is not in the document]
Comment (ry-animal) [reply src/components/transactions/ClaimRewards/ClaimRewardsModalContent.tsx]: removed from PR

File: src/modules/dashboard/DashboardTopPanel.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (ry-animal) [reply src/modules/dashboard/DashboardTopPanel.tsx]: Removed from PR
